src/main/scala/calc/Compiler.scala

```diff
@@ -74,6 +80,21 @@ object Compiler {
       case Literal(value) =>
         irt.DoubleLiteral(value)
 
+      case BinaryOp(op, rhs, lhs) =>
+        irt.BinaryOp(binaryOpMap(op), compileExpr(rhs), compileExpr(lhs))
+
+      case Let(name, value, body) =>
+        identMap.put(name, value)
```

Comment:
In fact, the very use of a mutable data structure is a code smell, *especially* as a global.
The compiler for this language can and should be written in a pure functional style.